staging/src/k8s.io/apiserver/plugin/pkg/audit/buffered/buffered_test.go

```diff
@@ -0,0 +1,290 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buffered
+
+import (
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	auditinternal "k8s.io/apiserver/pkg/apis/audit"
+	"k8s.io/apiserver/pkg/audit"
+)
+
+type options struct {
+	withWaitGroup bool
+	withChannel   bool
+}
+
+func newBufferedBackend(opts options) (*bufferedBackend, *fakeBackend) {
+	fakeBackend := newFakeBackend()
+	switch {
+	case opts.withChannel == true:
+		fakeBackend.ch = make(chan struct{})
+	case opts.withWaitGroup == true:
+		fakeBackend.wg = new(sync.WaitGroup)
+	default:
+	}
+	backend := NewBackend(fakeBackend)
+	return backend.(*bufferedBackend), fakeBackend
+}
+
+// waitForEmptyBuffer indicates when the sendBatchEvents method has read from the
+// existing buffer. This lets test coordinate closing a timer and stop channel
+// until the for loop has read from the buffer.
+func waitForEmptyBuffer(b *bufferedBackend) {
+	for len(b.buffer) != 0 {
+		time.Sleep(time.Millisecond)
+	}
+}
+
+func TestBufferedBackendBatchMaxEvents(t *testing.T) {
+	events := make([]*auditinternal.Event, defaultBatchMaxSize+1) // greater than max size.
+	for i := range events {
+		events[i] = &auditinternal.Event{}
+	}
+
+	stopCh := make(chan struct{})
```

Comment:
Just use `wait.NeverStop` instead of creating a new `stopCh` every time

Reply:
Can not, need to close the stopCh.

Reply:
Not in every test. In this, for example, you don't